### ISSUE TITLE
Latest waterfall disconnect spam

### ISSUE THREAD

**Issue by Cobwebster** [PaperMC/Waterfall#152]
We just created this simple gamemode on our network, if someone leaves from it .. it spams the console with connect/disconnect messages.. about 5k lines a second.

15:00:48 [INFO] [katieefaw|/XXX0XXXX] <-> ServerConnector [lobby2] has disconnected
15:00:48 [INFO] [katieefaw|/XXXXXXXXX] <-> ServerConnector [lobby] has connected

^-- That is what the entire log looks like, we've narrowed it down to when someone leaves our new gamemode. Any suggestions on tracking it down? (errors are in the bungee console) 

10.07 04:14:12 [Multicraft] Skipped 5968 lines due to rate limit (60/s)
10.07 04:14:13 [Multicraft] Skipped 6878 lines due to rate limit (60/s)
10.07 04:14:14 [Multicraft] Skipped 6436 lines due to rate limit (60/s)
10.07 04:14:15 [Multicraft] Skipped 6296 lines due to rate limit (60/s)
10.07 04:14:16 [Multicraft] Skipped 6078 lines due to rate limit (60/s)
10.07 04:14:17 [Multicraft] Skipped 4897 lines due to rate limit (60/s)

**Comment by Rushmead** [comment]
What plugins are you running on your waterfall instance?

**Comment by Cobwebster** [comment]
ViaVersion, litebans, MultiLobby <-- maybe the cause, contacted the dev. 
Serverlistplus & The plugin that lets you do /servername so /kitpvp

**Comment by Cobwebster** [comment]
We actually resolved it with the multilobby dev :)

**Comment by jamierocks** [comment]
Ok.

**Comment by Cobwebster** [comment]
Turns out it wasn't related to multilobby and the issue continues, we randomly have burts of like 20-50k logs of a random player disconnecting/connecting..

**Comment by fuzzybot** [comment]
Is the affected player protocol translated with ViaVersion in the proxy?

**Comment by Cobwebster** [comment]
We have ViaVersion running on our proxy, i'm unable to confirm if the person/people who trigger the spam join/leave are using a particular version though if thats what you mean. It seems to just pick people at random.

**Comment by fuzzybot** [comment]
Install a plugin that tracks the client version and find out. I have a
suspicion you are having issues with ViaVersion in the proxy.

**Comment by Cobwebster** [comment]
Okay so basically we narrowed down some information, we removed version support off a gamemode we have so its only 1.12 clients.

When someone disconnects this 1 gamemode it can trigger the spam, the gamemode is a survival/vanilla type so it literally has only like 6 plugins but we narrowed it down to that. I see no reason why the plugins on the server would cause issues so may be viaversion, does that info help? 1.12 client only gamemode still causing the issue. only thing that would ping bungee is DeluxeChat and viaversion

EDIT*

If they join another gamemode without leaving the proxy so like /survival it doesn't happen however if they disconnect from the server whilst on the gamemode it causes spam.
